### PR TITLE
Make TraverseContext readonly and pass via in

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Make `TraverseContext` readonly and pass via `in` ref (chonky struct)

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes (Modified existing for ReadOnlySpan)
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Trie/PatriciaTree.cs [hunk 1]
```diff
@@ -910,7 +914,15 @@ TrieNode secondExtension
             return traverseContext.UpdateValue;
         }
 
-        private static int FindCommonPrefixLength(Span<byte> shorterPath, Span<byte> longerPath)
+        private byte[] TraverseNext(in TraverseContext traverseContext, int extensionLength, TrieNode next)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Trie/PatriciaTree.cs:917]: Let's consolidate 2xTraverseNext methods

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Trie/PatriciaTree.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Trie/PatriciaTree.cs:909]: Done

**Comment by LukaszRozmej** [comment]
Did you synced/run a node with this changes?

**Comment by benaadams** [comment]
> Did you synced/run a node with this changes?

Seems happy?

```
2023-03-09 12:06:19.3782|Executing JSON RPC call eth_getBlockByHash with params [0xd5dd5e3ffc6dbacba6b59ce74544861e8c48da924ab27b32e188a4f572f26358, True]
2023-03-09 12:06:19.3945|Executing JSON RPC call eth_getBlockByHash with params [0x4281b334e1c3c1bcb428b3d42860637f0b97f73355882ed47b4e5842f1f63214, True]
2023-03-09 12:06:19.4253|Executing JSON RPC call eth_getBlockByHash with params [0x7091d23e2b4f33edb59c51c7d67a2de3a8daad8fb6dc96dffc7e81c3467eab53, True]
2023-03-09 12:06:19.4253|Executing JSON RPC call eth_getBlockByHash with params [0x6388339b7b5701c4ba6416f99f39f9899b435804abb534c70a451d684748a840, True]
2023-03-09 12:06:19.4579|Executing JSON RPC call eth_getBlockByHash with params [0x3aed05c6d4d04a70125cc8a7785740106d66a2da1399a53f0eb1e7d2fbb49bff, True]
2023-03-09 12:06:19.4787|Executing JSON RPC call eth_getBlockByHash with params [0x868c951992713c729f980e2bb7fe07597830582ca74bc2454faf2ab904941ce0, True]
2023-03-09 12:06:19.4968|Executing JSON RPC call eth_getBlockByHash with params [0xdfe6a5d855b9834bf309591b16ba930ff266fc1a75f4410e95901af8ae3512c1, True]
2023-03-09 12:06:19.5279|Executing JSON RPC call eth_getBlockByHash with params [0x11e0ea16e8aa2f5c90c250688bdb21dffe8d32a26ea40db39e011b35ae6e30b1, True]
2023-03-09 12:06:19.5545|Executing JSON RPC call eth_getBlockByHash with params [0xfc59f82be1ca67ec6f03d0c94ee5431931a437bdf2875aededd77765dc90a44c, True]
2023-03-09 12:06:19.5690|Executing JSON RPC call eth_getBlockByHash with params [0x7b683b87c09c88614060ed843c39c401a02e125cff92534ff358d8ef5b2147f7, True]
2023-03-09 12:06:19.5931|Executing JSON RPC call eth_getBlockByHash with params [0x43b9dda76dfffc6cfe94e9549b2bd40e4b289c8957d63e760cce8bde1e67b6d2, True]
2023-03-09 12:06:19.6162|Executing JSON RPC call eth_getBlockByHash with params [0x512aa1d2ae38dde18d36810269bfa42883253d0d3ba5789d3138e3bb0034c2b2, True]
2023-03-09 12:06:19.6348|Executing JSON RPC call eth_getBlockByHash with params [0x90177c2d6e7dee05b38ba16f4e36cb74442069253668bd5968c292c66b05abed, True]
2023-03-09 12:06:19.6795|Executing JSON RPC call eth_getBlockByHash with params [0x693dc6e5f34bc007d7f99f229dda75b6b1de3754fe524c29a7914f0068b11a8d, True]
2023-03-09 12:06:19.7066|Executing JSON RPC call eth_getBlockByHash with params [0x8349348d972f22338fcf5913e0059b60caa9228e2dc67480fda13e84128634fa, True]
2023-03-09 12:06:19.7344|Executing JSON RPC call eth_getBlockByHash with params [0xc84f42d3b015d42f894b5128baecad983eb5b370c2cfbd0a6b2fc93e65f6b658, True]
2023-03-09 12:06:24.4021|Executing JSON RPC call eth_getBlockByHash with params [0x70b3c5285d39681e5ae41905a3471f21b591ce12be07749252e4fa1cb03b5ab9, True]
2023-03-09 12:06:27.9185|Received a new payload: 16790542 (0x4e9145b44435b1b71fe16710faaf0166c307c776f5af2884c6b7ca509e02fe93)
2023-03-09 12:06:28.3393|Processed   16790542 |      419ms of  11,836ms, mgasps   73.34 total   61.37, tps  618.31 total  542.94, bps    2.39 total    2.00, recv queue 0, proc queue 0
2023-03-09 12:06:28.3393|Valid. Result of a new payload: 16790542 (0x4e9145b44435b1b71fe16710faaf0166c307c776f5af2884c6b7ca509e02fe93).
2023-03-09 12:06:28.4208|Received: ForkchoiceState: (HeadBlockHash: 0x4e9145b44435b1b71fe16710faaf0166c307c776f5af2884c6b7ca509e02fe93, SafeBlockHash: 0xc80a39a70c8dc832c2d802db471334540f01152808cf328ba975001edbbec167, FinalizedBlockHash: 0x0d62e24d9f947e35cf68a68e026f510c019633cd9d778bb69cbb36afa6cb892c)
2023-03-09 12:06:28.4208|FCU - block 16790542 (0x4e9145...02fe93) was processed.
2023-03-09 12:06:28.4208|Block 0x4e9145b44435b1b71fe16710faaf0166c307c776f5af2884c6b7ca509e02fe93 was set as head.
2023-03-09 12:06:28.4208|Valid. Request: ForkchoiceState: (HeadBlockHash: 0x4e9145b44435b1b71fe16710faaf0166c307c776f5af2884c6b7ca509e02fe93, SafeBlockHash: 0xc80a39a70c8dc832c2d802db471334540f01152808cf328ba975001edbbec167, FinalizedBlockHash: 0x0d62e24d9f947e35cf68a68e026f510c019633cd9d778bb69cbb36afa6cb892c) .
2023-03-09 12:06:28.5488|Executing JSON RPC call eth_syncing with params []
2023-03-09 12:06:30.3163|Peers | with known best block: 167 | all: 169 |
2023-03-09 12:06:39.9632|Received a new payload: 16790543 (0xb23aeed9ecbcdedae06050967a65222ea4cdbc86f84f16c3478ceea4a7087392)
2023-03-09 12:06:40.4630|Processed   16790543 |      497ms of  12,124ms, mgasps   64.48 total   61.40, tps  581.06 total  543.28, bps    2.01 total    2.00, recv queue 0, proc queue 0
2023-03-09 12:06:40.4630|Valid. Result of a new payload: 16790543 (0xb23aeed9ecbcdedae06050967a65222ea4cdbc86f84f16c3478ceea4a7087392).
2023-03-09 12:06:40.5354|Received: ForkchoiceState: (HeadBlockHash: 0xb23aeed9ecbcdedae06050967a65222ea4cdbc86f84f16c3478ceea4a7087392, SafeBlockHash: 0xc80a39a70c8dc832c2d802db471334540f01152808cf328ba975001edbbec167, FinalizedBlockHash: 0x0d62e24d9f947e35cf68a68e026f510c019633cd9d778bb69cbb36afa6cb892c)
2023-03-09 12:06:40.5354|FCU - block 16790543 (0xb23aee...087392) was processed.
2023-03-09 12:06:40.5354|Block 0xb23aeed9ecbcdedae06050967a65222ea4cdbc86f84f16c3478ceea4a7087392 was set as head.
2023-03-09 12:06:40.5354|Valid. Request: ForkchoiceState: (HeadBlockHash: 0xb23aeed9ecbcdedae06050967a65222ea4cdbc86f84f16c3478ceea4a7087392, SafeBlockHash: 0xc80a39a70c8dc832c2d802db471334540f01152808cf328ba975001edbbec167, FinalizedBlockHash: 0x0d62e24d9f947e35cf68a68e026f510c019633cd9d778bb69cbb36afa6cb892c) .
2023-03-09 12:06:40.5491|Executing JSON RPC call eth_syncing with params []
2023-03-09 12:06:45.4096|Peers | with known best block: 171 | all: 173 |
2023-03-09 12:06:45.4096|Allocated sync peers 171(173)/250
2023-03-09 12:06:46.5839|Received: ForkchoiceState: (HeadBlockHash: 0xb23aeed9ecbcdedae06050967a65222ea4cdbc86f84f16c3478ceea4a7087392, SafeBlockHash: 0xa3a8b8148c0723716d243353e99436abeb50f751b04807117115b449e74808f4, FinalizedBlockHash: 0xc80a39a70c8dc832c2d802db471334540f01152808cf328ba975001edbbec167)
2023-03-09 12:06:46.5839|FCU - block 16790543 (0xb23aee...087392) was processed.
2023-03-09 12:06:46.5839|Valid. Request: ForkchoiceState: (HeadBlockHash: 0xb23aeed9ecbcdedae06050967a65222ea4cdbc86f84f16c3478ceea4a7087392, SafeBlockHash: 0xa3a8b8148c0723716d243353e99436abeb50f751b04807117115b449e74808f4, FinalizedBlockHash: 0xc80a39a70c8dc832c2d802db471334540f01152808cf328ba975001edbbec167) .
2023-03-09 12:06:47.1984|Received a new payload: 16790544 (0x2e6aafcc91dea5023c086bd1a2c3fc252bf95bb0d976e5a6dc6db285204e5272)
2023-03-09 12:06:47.5349|Processed   16790544 |      333ms of   7,072ms, mgasps  104.68 total   61.65, tps 1093.85 total  546.53, bps    3.01 total    2.01, recv queue 0, proc queue 0
2023-03-09 12:06:47.5349|Valid. Result of a new payload: 16790544 (0x2e6aafcc91dea5023c086bd1a2c3fc252bf95bb0d976e5a6dc6db285204e5272).
2023-03-09 12:06:48.5888|Received: ForkchoiceState: (HeadBlockHash: 0x2e6aafcc91dea5023c086bd1a2c3fc252bf95bb0d976e5a6dc6db285204e5272, SafeBlockHash: 0xa3a8b8148c0723716d243353e99436abeb50f751b04807117115b449e74808f4, FinalizedBlockHash: 0xc80a39a70c8dc832c2d802db471334540f01152808cf328ba975001edbbec167)
2023-03-09 12:06:48.5888|FCU - block 16790544 (0x2e6aaf...4e5272) was processed.
2023-03-09 12:06:48.5888|Block 0x2e6aafcc91dea5023c086bd1a2c3fc252bf95bb0d976e5a6dc6db285204e5272 was set as head.
2023-03-09 12:06:48.5888|Valid. Request: ForkchoiceState: (HeadBlockHash: 0x2e6aafcc91dea5023c086bd1a2c3fc252bf95bb0d976e5a6dc6db285204e5272, SafeBlockHash: 0xa3a8b8148c0723716d243353e99436abeb50f751b04807117115b449e74808f4, FinalizedBlockHash: 0xc80a39a70c8dc832c2d802db471334540f01152808cf328ba975001edbbec167) .
2023-03-09 12:06:52.5485|Executing JSON RPC call eth_syncing with params []
2023-03-09 12:06:56.8615|Executing JSON RPC call eth_chainId with params []
2023-03-09 12:06:56.8615|Executing JSON RPC call eth_getBlockByNumber with params [latest, False]
```

**Comment by benaadams** [comment]
Running for 3 hours now, still seems happy

**Pull request merged**